2-repositories/scala-repo-server/src/main/scala/com.snowplowanalytics.iglu.server/package.scala

```diff
@@ -17,15 +17,17 @@ package com.snowplowanalytics.iglu
 // Json schema
 import com.github.fge.jsonschema.core.report.ProcessingMessage
 
+// Scala
+import scala.util.matching.Regex
+
 // Scalaz
 import scalaz._
-import Scalaz._
 
 package object server {
   type ValidatedNel[A] = ValidationNel[ProcessingMessage, A]
 
-  val VendorPattern = "[a-zA-Z0-9-_.]+".r
-  val NamePattern = "[a-zA-Z0-9-_]+".r
-  val FormatPattern = "[a-zA-Z0-9-_]+".r
-  val VersionPattern = "[0-9]+-[0-9]+-[0-9]+".r
+  val VendorPattern: Regex = "[a-zA-Z0-9-_.]+".r
+  val NamePattern: Regex = "[a-zA-Z0-9-_]+".r
+  val FormatPattern: Regex = "[a-zA-Z0-9-_]+".r
+  val VersionPattern: Regex = "[0-9]+-[0-9]+-[0-9]+".r
```

Comment:
Since akka-http does [not allow](https://doc.akka.io/docs/akka-http/current/routing-dsl/path-matchers.html) PathMatchers, in case they are regex, to contain more than one capture group, it isn't possible to directly make use of [schemaVerFullRegex](https://github.com/snowplow/iglu/blob/master/0-common/scala-core/src/main/scala/com.snowplowanalytics.iglu/core/SchemaVer.scala#L58) so I cancelled capture groups and made use of the same regex.

Reply:
Updated `VersionPattern` by making partial use of [`schemaUriRegex`](https://github.com/snowplow/iglu/blob/master/0-common/scala-core/src/main/scala/com.snowplowanalytics.iglu/core/SchemaKey.scala#L45). I couldn't use [`schemaVerFullRegex`](https://github.com/snowplow/iglu/blob/master/0-common/scala-core/src/main/scala/com.snowplowanalytics.iglu/core/SchemaVer.scala#L58) directly since it contains multiple capture groups which akka-http doesn't allow for a PathMatcher regex.